provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -441,38 +443,94 @@ func (c *Client) DeleteDatastoreFile(ctx context.Context, datastorePath string)
 	return nil
 }
 
-func (c *Client) destroyVM(
-	ctx context.Context,
-	vm *object.VirtualMachine,
-	taskWaiter *taskWaiter,
-) error {
-	task, err := vm.Destroy(ctx)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	_, err = taskWaiter.waitTask(ctx, task, "destroying VM")
-	return errors.Trace(err)
+// rootDiskB converts the root disk constraint (which uses megabytes for its unit) to bytes
+func rootDiskB(diskMiB uint64) uint64 {
```

Comment:
These might be better named for what they do, rather than that they're used on the root disk size - megabytesToB and megabytesToKiB?

Reply:
Makes sense, fixed.